extensions/objects/protractor.js

```diff
@@ -21,6 +21,19 @@
 // implement a setValue() function to which a single argument can be sent
 // that will completely determine the object.
 
+var AlgebraicExpressionEditor = function(elem) {
+  return {
+    setValue: async function(text) {
+      await elem.click();
+      var algebraicInputElem = element(by.css('.guppy_active'));
```

Comment:
Can we use a `protractor-test-*` tag here? Even if you have to drill down to a child element from the `protractor-test-*` tag like `element(by.css(protractor-test-...)).element(...).element(...)...`, including a `protractor-test-*` tag makes it easier to find the relevant part of the HTML when we're debugging tests

Reply:
The `guppy_active` class is added dynamically by the third-party lib so I can't change that class name. Also, at a given instant, there would be multiple guppy divs present so I can't query based on the guppy-div class name since only one of them will be active (only one of them will have the class `guppy_active`) which is why we try to find that particular div by querying for `guppy_active` classname.

Reply:
Say the `guppy-div` element has the class `protractor-test-guppy-div`. Then couldn't you do the following?

```js
element(by.css(protractor-test-guppy-div)).element(by.css('.guppy_active'))
```

Reply:
Yeah, but at a given moment there could be multiple divs in the document that have the class `protractor-test-guppy-div` and only one of them would contain `.guppy_active`. Protractor would grab the first div but that might not be the one that contains the active class.

Reply:
You can loop over the elements with `protractor-test-guppy-div` and get the one with `.guppy_active`. Here's an example of this pattern:

https://github.com/oppia/oppia/blob/3317ab53340a3b3161cdb65ba255e891dcd3874b/core/tests/protractor_utils/LibraryPage.js#L56

Also for future reference, this is one of the patterns described on the [wiki page](https://github.com/oppia/oppia/wiki/End-to-End-Tests#independence), where we have a link to that example if you ever need it again

Reply:
That actually inspired a solution! I realized that the active div will be the one that's created last so I just picked the last div and it works as expected. I've now used `protractor-test-guppy-div`, thanks!